### DISCUSSION TITLE
Electrode localization with different probes than Neuropixel

### DISCUSSION THREAD

**Discussion by titomma** [JingyiGF/HERBS#19] [Ideas]
Hi! Great tool! I found it on the best moment because I am currently studying how I can perform electrode localization from my implant. In this moment, I wish to use Neuropixel probes because I would totally benefit from HERBS. But since I am using a different type of silicon probes, do you think it would be possible to customize the probes for the localization rather than using just the Neuropixel design or the tetrodes? Maybe, if the users could insert some information about the probes they are using (as the number of shafts, shaft length, the number of electrodes, their design and pitch for example), this would be possible. 
I am curious to know whether this is possible and if you may plan to implement it :)

**Comment by JingyiGF** [comment]
Hi, thanks for the request. This is a very interesting topic for me and I think it is totally doable. To make this funcitonality feasible, it will be great help if you let me know what kind of probes you would like to insert and what are the features you are looking for. I have only seen NP 1.0 and 2.0, and no experiences with other probes.

**Reply by titomma** [reply]
Thank you a lot! I am still interested to know in which structure each electrode on the probe is and I am struggling to reconstruct the probe on my sections. I am currently using 32 channel silicon probes where the electrodes are linearly displaced. I share with you the link of my [probes](https://www.atlasneuro.com/en/products/32-channel-probes/1113). Here there are all the specs of this particular probe design, but there are many more possible designs out there :)

**Reply by JingyiGF** [reply]
Thanks for the info. As you might know from other issues or discussions that I am not a neuroscientist, so I may not get the correct information without any explanation. For example, what is a shaft and a pitch in the link you gave? Same word in different context can mean different things ;-) I would guess the black dots are the electrodes? the tip is shown on the top, what is the distance from the actual physical tip to the first electrode? 'electrode length' - from where to where?

**Reply by titomma** [reply]
Yes, sorry, I am also quite new in the field and I hope to give you the correct definitions. For what I understood, the shaft is the whole probe shank and this is long 6mm. If there are more shafts (not in this case), the shaft pitch would be the distance between the two shanks. As you say, the electrodes are the black dots on the picture and the electrode pitch in this case should be the distance between the centers of 2 neighbouring electrodes. Then, the electrode length (and for me this is a bad definition, easily leading to misunderstanding) should be the whole electodes' area spanning from the first to the 32nd electrode. In the end, I don't know which is the actual physical distance from the first electrode and the tip of the probe, but it is something that I can try to ask to the seller! I made a small drawing with this description 
![probe_description](https://user-images.githubusercontent.com/66744022/196389608-c4bfff27-cbba-4d42-83a4-552d4eea1d30.png)
If you need any more information I can try to look for and pass them to you :)

**Reply by JingyiGF** [reply]
Thanks for the clear explanation! I put your request on the list. before your request are customer defined atlas processor and probe visualisations in 2D window. I will come back to you when I start implementing this topic. Please let me know the tip-electrode distance when you have it and other specifications you would like to have :-) many thanks for the clear illustration again 🤩

**Reply by titomma** [reply]
Dear Jingyi, here the last bit of information: the probe producer told me that the distance of the tip from the center of the first electrode is 60um.
Cannot wait to have news from you :)

**Reply by JingyiGF** [reply]
Hi, linear silicon probe is added in HERBS in GitHub version. Will update you when I update HERBS on PyPI.

**Reply by JingyiGF** [reply]
Hi, sorry for the late reply. I am now update code both on GitHub and PyPI. and also the relevant tutorials https://github.com/JingyiGF/HERBS/tree/main/Tutorial/Probe_Related. Please let me know if anything is not clear or any suggestion and idea to make it better :-) 

Thank you for this interesting request that leads HERBS to a new world. Wish you a happy holiday season and see you in 2023!

**Comment by Irina8880** [comment]
Dear @JingyiGF 

My MEA is an array with a fixed diameter and recording units are located in a circular plane. The depth is also fixed by the MEA design (similar to the UTAH array). I need to estimate what areas and layers my device span, depending on an insertion angle and the center. Or opposite - to estimate the insertion angle and center depending on what layers and areas I want to record. 

![image](https://github.com/JingyiGF/HERBS/assets/134515673/bbce0435-fe2f-4ab8-8028-a1078abdb67e)
(Basically, I want to plan where the tips of my wires are.)
Here is the paper describing the device. 
https://www.science.org/doi/10.1126/science.abf4588

I'm wondering, if would you be interested in developing this. 
Thank you!

Best wishes,
Irina

**Comment by JingyiGF** [comment]
Hi, sorry for the late reply if any. I am super busy recently so i didn’t manage to reply properly to many messages. But I am slowly update the code and instructions. 

What’s the difference with your probe compare to silicon probe? Please read the update log and tutorial on HERBS GitHub page first and let me know if you find any useful info. If not, we can have a discussion about it. 

Best,
Jingyi

**Comment by Marekthegeck** [comment]
Hi @JingyiGF 

Many thanks for your fantastic work first of all! I started to work with HERBS only last week, but it already became my favorite tool to register brain sections to atlas! Now, I have a request that best fits into this discussion, and I'm very sorry if that was addressed somewhere else already and I've missed that. 

Instead of ephys probes, I implant GRIN lenses for functional imaging of deep brain tissue. These lenses are of a simple tubular geometry and have a diameter of 500-1000 microns. To make use of all HERB functions, especially the 3d reconstructions, I would need a "probe" designed to reflect the lens. I played around with the probe designer you implemented here, however to get it right, it would be best to have the option to design a tubular probe with various diameters (no need for sites, columns etc) and then simply tell HERBS if the current section reflects the center, caudal or rostral end of the "probe" (the lenses are recovered from the brains at experimental endpoints). Usually, a virus is used to be able to image cellular activity, but for this HERBS has everything on board already. 

What do you think, would something in that direction be possible? No rush here, but I do think HERBS would gain a hole lot of new friends through this :) 

Cheers
Marek

**Reply by JingyiGF** [reply]
Hi Marek, this sounds interesting. Could you please provide more info on "tubular probe"? what is it? how does it work? what is the architecture? any different designs for different tubular probe? and difference between tubular probe and others?

**Reply by Marekthegeck** [reply]
Hi Jingyi,
Many thanks for your fast response! The tubular "probe" is not actually a probe, but a lens that is used to image cells that express a functional fluorophore underneath the lens (like a calcium indicator e.g. GCaMP) The lens allows access to deep brain structures using various types of microscopes such as simple widefield or 2photon. So to decipher the imaging plane from histological sites, you need information about where the fluorophore is expressed (which is already implemented in HERBS) and additionally where the lens was placed. Here, it is not as important to know the actual brain regions where the lens was (as it is for the neuropixel probes) because the data we collect derives actually from the cells underneath the lens, but simply the coordinates from where to where the lens spans in a 3d space as a-p,d-v,m-l. Of course, it is possible to take the most apical and the most posterior section of where the lens was. But it would be simpler and smoother to have a "probe" object with a tubular geometry, especially for a 3d reconstruction. 

Here is an example section and image of the lens to make things more clear:

![lens_example](https://github.com/JingyiGF/HERBS/assets/102272302/07c4f48a-8236-4c6e-a390-fda60808e98b)

![image](https://github.com/JingyiGF/HERBS/assets/102272302/0f6d38a1-7ba1-476d-a941-4af6136327b4)


 Usually, these lenses are all simple tubular with diameters ranging from 500-1000 microns. In some rare cases, they may have a prism at the end to shift the imaging plane.

**Reply by JingyiGF** [reply]
Thank you for the clear explanation and images for illustration. Please let me rephrase your request. So you would like to have a way to reconstruct a tubular probe using the histological images and visualize the probe in 3D as a cylinder instead of a line. Is this correct? 

This sounds very interesting technically. Would it be possible to share the whole file with me? you can find my personal email on the GitHub page. To make this happen, I will need a lot more info from you, we can discuss through emails if suits you.
